library/core/tests/num/const_from.rs

```diff
@@ -0,0 +1,25 @@
+#[test]
+fn from() {
+    use core::convert::TryFrom;
+    use core::num::TryFromIntError;
+
+    // From
+    const FROM: i64 = i64::from(1i32);
+    assert_eq!(FROM, 1i64);
+
+    // From int to float
+    const FROM_F64: f64 = f64::from(42u8);
```

Comment:
cc @rust-lang/wg-const-eval this is an interesting edge case around floats. We forbid floats in const fn, but those checks  only happen at the declaration site, and libcore uses the feature gate that allows them. So now we can invoke const fns (only within items, not within other const fn) that work with floats.

Reply:
Yeah, we should really work towards getting rid of "things that are allowed in `const` but forbidden in `const fn`"...